### DISCUSSION TITLE
Issue with New Volumes Not Automatically Added to Scheduled CronJobs for Backups

### DISCUSSION THREAD

**Discussion by othapall** [longhorn/longhorn#9920] [General]
We have configured a Longhorn CronJob to back up volumes associated with the default RWO (ReadWriteOnce) storage class. While the CronJob works effectively for existing volumes, any new volumes created under the same storage class are not automatically included in the backup schedule.

Steps to Reproduce:

Create a Longhorn CronJob to schedule backups for volumes under the default storage class (RWO).
Verify that backups are successfully taken for the existing volumes.
Create new volumes in the default storage class.
Observe that the new volumes are not automatically added to the CronJob's backup schedule.
Expected Behavior:
Newly created volumes in the default storage class should automatically be included in the existing CronJob's backup schedule.

Actual Behavior:
New volumes are excluded from the CronJob schedule, requiring manual intervention to include them.

Environment Details:

Longhorn version: 1.6.3
Kubernetes version: 1.30
Storage class configuration:
kind: StorageClass
apiVersion: storage.k8s.io/v1
metadata:
  name: default-rwo
  annotations:
    storageclass.kubernetes.io/is-default-class: "true"
provisioner: driver.longhorn.io
allowVolumeExpansion: true
parameters:
  numberOfReplicas: {{ quote .Values.longhorn_storage.replicas }}
  staleReplicaTimeout: "30"
  fromBackup: ""
  dataLocality: best-effort
  recurringJobs: '[
    {
      "name":"snap",
      "task":"snapshot",
      "cron":"0 */2 * * *",
      "retain":12
    },
    {
      "name":"backup",
      "task":"backup",
      "cron":"0 0 * * *",
      "retain":14
    }
  ]'

Request for Support:
Is this the expected behavior, or is there a way to configure Longhorn so that new volumes are automatically added to the backup schedule? If not, would it be feasible to enhance this feature in a future release?

**Comment by mantissahz** [comment]
track on https://github.com/longhorn/longhorn/issues/9921
